### ISSUE TITLE
[FEATURE REQUEST] ability to send the Module struct to the LLVM lib

### ISSUE THREAD

**Issue by einsJannis** [cdisselkoen/llvm-ir#20]
As pointed out in the README: this library is currently unable to send the Rust-y and safe `Module` struct back to the `LLVM` library, and thus can't be used for LLVM IR generation nor for compilation of LLVM IR.

I think this library would benefit from the ability to send the module struct back to the LLVM library.

**Comment by cdisselkoen** [comment]
I agree that this would be a great feature.  But it's a very heavy lift and I currently have no plans to take that on, sorry.  If you're interested in helping develop this, feel free to make a PR, maybe starting with a very simple case (only handling some small subset of LLVM features or something).

**Comment by cdisselkoen** [comment]
Related to #46: solving either of these individually would imply a solution for the other.
